ci/bin/build-and-test

```diff
@@ -32,8 +48,10 @@ puts "=" * 80
 
 raise "Missing config YML file" if config_path && !(File.exist?(config_path) && File.file?(config_path))
 
-puts "Installing latest rails gem"
-system "gem install rails --no-document"
+puts "Installing latest compatible rails gem"
+install_output = `gem install rails --version '#{rails_version_specifier}' --no-document`
+rails_version = install_output.split("\n").first.strip.sub("Successfully installed rails-", "")
+puts "Using rails version #{rails_version}"
```

Comment:
We want to run CI against our supported Rails `major.minor` with the latest `patch` number.